middlewares/osio/auth.go

```diff
@@ -120,9 +137,41 @@ func extractToken(auth string) (string, error) {
 	return auths[len(auths)-1], nil
 }
 
-func cacheKey(token string) string {
+func cacheKey(token, service string) string {
 	h := sha256.New()
-	h.Write([]byte(token))
+	key := fmt.Sprintf("%s_%s", token, service)
```

Comment:
now cache key is combination of token+service where token is user OSIO token and service could be api/metrics.